src/newTest/java/org/opensearch/test/framework/cluster/NestedValueMap.java

```diff
@@ -0,0 +1,518 @@
+/*
+ * Copyright 2021-2022 floragunn GmbH
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * 
+ */
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.test.framework.cluster;
+
+import java.io.File;
+import java.lang.reflect.Array;
+import java.net.URI;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import com.google.common.collect.MapMaker;
+
+public class NestedValueMap extends HashMap<String, Object> {
```

Comment:
I don't see where this is used

Reply:
Seems like this still hasn't been removed

Reply:
Working on it, will update you again.

Reply:
`TestSecurityConfig` has been now refactored to use `ToXContentObject` instead of `NestedValueMap`. The `NestedValueMap` class has been removed now.